### ISSUE TITLE
support for transient queues?

### ISSUE THREAD

**Issue by davewongillies** [bodgit/graphite-amqp-tools#13]
Can graphite-amqp-tools handle transient queues? I've attempted to connect to a transient queue I get the following error:

```

Jul 11 04:05:26 server01 graphite-dequeue[27914]: server channel error 406, message: PRECONDITION_FAILED - parameters for queue 'queue01' in vhost 'vhost01' not equivalent
Jul 11 04:05:26 server01 graphite-dequeue[27914]: fatal: amqp_queue
```

**Comment by bodgit** [comment]
It currently creates durable queues which I think is "non-transient", unless that's yet another flag.

I think I did durable queues as the safest option, but I can understand why someone might want to make them transient for performance reasons. Probably just needs exposing a configuration option and tweaking the queue create code, I'll have a look.

As an aside, I'd be interested to know what sort of performance you're getting.

**Comment by bodgit** [comment]
I've hopefully committed a fix for this in 8439f9f let me know if this sorts it for the dequeue daemon.

If you're using the enqueue daemon, then this will still be marking the messages as persistent, not sure if that affects things or not.

**Comment by davewongillies** [comment]
So I've finally gotten a chance to test this out in anger with the dequeue daemon and its working perfectly, thanks!

bodgit said:

> If you're using the enqueue daemon, then this will still be marking the messages as persistent, not sure if that affects things or not.

I'm not normally using the enqueue daemon, my metric flow goes collectd -> rabbitmq -> dequeue -> graphite, so I'm all good.

bodgit said:

> As an aside, I'd be interested to know what sort of performance you're getting.

I've got a cluster in AWS Ireland, which metrics get published to rabbitmq queues locally, with the dequeue running in AWS Singapore, publishing to the graphite server there, so along with the usual AWS perf issues, I've got latency and other general Internet routing crappiness thrown into the mix.

There's a reasonable amount of pressure on the rabbits (they're sharing instances with some other software ATM). Due to some changes in our infrastructure, this is having a knock on affect on performance and slowing down the pulling metrics out of the rabbitmq queue.

So, with that in mind, on a durable queue, this morning I was managing to pull messages off the queue at roughly 2000-3000 messages/sec. This was with 4 copies of dequeue running. With a transient queue, 4 copies of dequeue I was seeing it top out at around 8000 messages/sec.

I was running multiple copies of dequeue since I found (this morning atleast) that each dequeue topped out roughly at 1000 (durable) and 2000 (transient) per second.

I don't know exactly why this was the case this morning though as yesterday I was consuming off a durable queue at +4000 messages per second on the exact same server.
